substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -336,8 +336,14 @@ impl<T: Config> Pallet<T> {
 			return None
 		}
 
-		let dest = Self::payee(StakingAccount::Stash(stash.clone()));
-		let maybe_imbalance = match dest {
+		let dest = match Self::payee(StakingAccount::Stash(stash.clone()))
+			.defensive_proof("payee must exist for a ledger requesting a payout")
```

Comment:
Would be good to have an integration test to ensure `RewardDestination` is never `None`.

Reply:
Added, thanks!